libsupport/include/katana/HostAllocator.h

```diff
@@ -0,0 +1,37 @@
+#pragma once
+
+#include <cstdlib>
+#include <cstring>
+
+#include "katana/config.h"
+
+namespace katana {
+
+enum class MemoryPinType { Swappable = 0, Pinned = 1 };
+
+class BaseHostAllocator {
+public:
+  virtual void* Malloc(const size_t n_bytes) const = 0;
+  virtual void* Calloc(const size_t n_items, const size_t item_size) const = 0;
+  virtual void* Realloc(void* ptr, const size_t n_bytes) const = 0;
+  virtual void Free(void* ptr) const = 0;
+  virtual MemoryPinType GetType() const = 0;
+  virtual ~BaseHostAllocator() {}
+};
+
+class SwappableHostAllocator : public BaseHostAllocator {
```

Comment:
Add a virtual destructor and put the implementation in the cpp file to pin the vtable to a single translation unit.